Port site from 11ty to Jekyll, move content back into markdown (not JSON)

jake-low opened the pull request:
This is intended to simplify contributing to the docs. Docs are now plain markdown files, rather than templates whose actual content lives in string values in a JSON file.

Translating docs can be done in two ways:
1. Using Transifex, as before. Each page is now a separate Transifex resource. Paragraphs, headings, and other chunks of each markdown file become individual translation strings in Transifex, and the translated values are combined back into a markdown document when they are pulled.
2. By simply creating a new markdown document for the new language and translating it in any text editor.

I fetched and imported the existing translations from Transifex, so there are a handful of translated docs already in the repository. However, for now I've restricted the production site to only display languages where a large % of the docs have been translated and reviewed (i.e. only English). You can uncomment the additional languages in `_config.yaml` to see them in development mode, and try out the new language switcher. Once more translations are available we can enable these languages in production.

The organization of the docs is mostly unchanged, with a few small exceptions:
- I made "Screencasts" into a section rather than a separate top level category, simplifying the sidebar.
- I did not port over the "Releases" section. The current release version is linked in the footer on maproulette.org, and for a list of other releases I think it's better to direct people to Github.

jake-low committed:
Port site from 11ty to Jekyll, move content back into markdown (not JSON)

This is intended to simplify contributing to the docs. Docs are now
plain markdown files, rather than templates whose actual content lives
in string values in a JSON file.

Translating docs can be done in two ways:
1. Using Transifex, as before. Each page is now a separate Transifex
   resource. Paragraphs, headings, and other chunks of each markdown
   file become individual translation strings in Transifex, and the
   translated values are combined back into a markdown document when
   they are pulled.
2. By simply creating a new markdown document for the new language and
   translating it in any text editor.

jake-low commented:
BTW the site builds in 1.1 seconds now. 🙂 

Here is what an example page looks like. The look and feel is largely unchanged from the eleventy version, though I tried to keep the CSS simple rather than emulate the old layout exactly.

The language picker in the top right shows how many languages the current page is available in. If it's only available in one language then the picker is replaced with a "Help translate!" link.

![Screenshot 2025-03-10 at 11-24-11 Challenge Comments MapRoulette Docs](https://github.com/user-attachments/assets/782f705a-9eb2-467c-80cb-4197f2b17247)

jake-low commented:
I think this is ready for review now. Once it's merged I'll need to do a bit of follow-up work to reconfigure netlify and do a `tx push` with the new resources.

mvexel reviewed (approved):
Looks great to me! Is http://127.0.0.1:4000/en-us/documentation/translate-docs/ up to date with the new setup?

mvexel committed:
lockfile update
when I installed locally (ruby 3.4.1, gem 3.6.6) and installed the dependencies, the lockfile was updated a bit. Not sure if relevant, feeel free to ignore..

jake-low committed:
Update translation guidance to describe new workflow

jake-low commented:
Good thought, thanks! I've updated the translation guidelines. Translators can still use Transifex as before, but they can also translate a Markdown file in their text editor and submit it via Pull Request; I added that info to the docs.

jake-low committed:
Tweak contributing instructions

jake-low committed:
Use latest ruby/setup-ruby action instead of pinning

jake-low committed:
Replace CoC file with a link to the authoritative document

jake-low commented:
Merging this now. There will be a brief (<5m) outage while I switch the DNS records over from Netlify to GitHub Pages.

jake-low commented:
Nevermind, there was no downtime (Cloudflare DNS updates commit super fast). Changes are now live.
